.github/workflows/release.yaml

```diff
@@ -1,32 +1,78 @@
 name: Release
 
 on:
+  schedule:
+    # Run on every first day of the month
+    - cron: '0 0 1 * *'
   push:
     tags:
       - '**'
 
 jobs:
 
-  release:
+  changelog:
     runs-on: ubuntu-latest
     steps:
-
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Create Package
-        run: tar -czf forgit-${{github.ref_name}}.tar.gz --exclude LICENSE --exclude README.md *
-
       - name: Generate Changelog
         id: changelog
         run: .github/scripts/changelog.sh > CHANGELOG.md
+        # When the script returns with an error, it indicates that we don't have
+        # any public changes to be released. This is expected behavior, so we
+        # quit the workflow successfully in this case.
         continue-on-error: true
 
+      - name: Upload Changelog
+        uses: actions/upload-artifact@v4
+        if: steps.changelog.outcome == 'success'
+        with:
+          name: changelog
+          path: CHANGELOG.md
```

Comment:
Is the uploading and downloading the changelog necessary for not tagging without public changes?

If not, could we separate it into it's own PR? If it's necessary, all good to keep it here.

Reply:
Changelog creation and release publishing are two different jobs now. I find this useful to make visible at first glance that only the changelog has run and release was skipped in the GitHub actions view.
Uploading and downloading artifacts is the way to share files between jobs, so it is required for this change. Branching this out into a dedicated PR does not make sense to me, because we do the complete change for the single purpose of preventing the tag.